### DISCUSSION TITLE
Purpose of page reloading

### DISCUSSION THREAD

**Discussion by discoverlance-com** [freshframework/fresh#302] [Q&A] [answered]
As described in the fresh documentation it ships with zero js bundles but I tried running the application (using main.ts). It did work alright but I can see in my network tab and in the sources tab that there's a `refresh.js` that's contsantly running and seems to be sending requests to the server at certain intervals.

What's the purpose of this script as fresh is supposed to ship with zero js in the client so I am not sure if it's an error on my side or it's by design.
The script when viewed in the sources has this content
```
let reloading = false; const buildId = "some-random-characters here"; 
new EventSource("/_frsh/alive").addEventListener("message", (e) => { 
  if (e.data !== buildId && !reloading) 
     { reloading = true; location.reload(); } 
});
```

**Comment by lucacasonato** [comment, marked as answer]
@Lance1997 This script is only shipped during development (when `DENO_DEPLOYMENT_ID` is unset). It reloads the page automatically when you change some code on the server. This is commonly referred to as hot refresh.
